src/main/java/com/minecolonies/coremod/entity/ai/mobs/AbstractArcherBarbarian.java

```diff
@@ -0,0 +1,309 @@
+    private static final int                              DIFFICULTY_SUBTRACTER = 14;
+    private static final int                              DIFFICULTY_MULTIPLIER = 4;
+    private static final int                              BOW_DAMAGE            = 20;
+    private static final int                              BOW_DAMAGE_HARD       = 40;
+    private static final double                           GRAVITY_MULTIPLIER    = 0.20000000298023224D;
+    private static final double                           ENTITY_Y_OFFSET       = -0.6D;
+    private static final float                            ARROW_VELOCITY        = 1.6F;
+    private static final float                            SOUND_MULTIPLIER      = 0.4F;
+    private static final float                            SOUND_ADDER           = 0.8F;
+    private static final float                            ENTITY_EYE_HEIGHT     = 1.74F;
+    private static final float                            RANDOM_FLOAT_EQUATER  = 0.55F;
+
+    /**
+     * Defines the default values for the Entity's attributes.
+     */
+    private static final double FOLLOW_RANGE          = 35.0D;
+    private static final double MOVEMENT_SPEED        = 0.2D;
+    private static final double ATTACK_DAMAGE         = 1.0D;
+    private static final double ARMOR                 = 2.0D;
+    private static final double BARBARIAN_BASE_HEALTH = 25;
+
+    /**
+     * Defines the default values for the various AI Task's properties
+     */
+    private static final int   PRIORITY_ONE       = 1;
+    private static final int   PRIORITY_TWO       = 2;
+    private static final int   PRIORITY_THREE     = 3;
+    private static final int   PRIORITY_FOUR      = 3;
+    private static final int   PRIORITY_FIVE      = 5;
+    private static final int   PRIORITY_SIX       = 6;
+    private static final float MAX_WATCH_DISTANCE = 8.0F;
+
+    private final EntityAIAttackMelee aiAttackOnCollide = new EntityAIAttackMelee(this, 1.2D, false)
+    {
+        /**
+         * Resets the task
+         */
+        public void resetTask()
+        {
+            super.resetTask();
+            AbstractArcherBarbarian.this.setSwingingArms(false);
+        }
+
+        /**
+         * Execute a one shot task or start executing a continuous task
+         */
+        public void startExecuting()
+        {
+            super.startExecuting();
+            AbstractArcherBarbarian.this.setSwingingArms(true);
+        }
+    };
+
+    public AbstractArcherBarbarian(final World worldIn)
+    {
+        super(worldIn);
+        this.setSize(ENTITY_WIDTH, ENTITY_HEIGHT);
+    }
+
+    protected void initEntityAI()
+    {
+        this.tasks.addTask(PRIORITY_ONE, new EntityAISwimming(this));
+        this.tasks.addTask(PRIORITY_TWO, new EntityAIRestrictSun(this));
+        this.tasks.addTask(PRIORITY_THREE, new EntityAIAvoidEntity(this, EntityWolf.class, 6.0F, 1.0D, 1.2D));
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 6.0F to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 1.2D to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)